js/client.js

```diff
@@ -1636,19 +1647,42 @@
 			document.title = room.title ? room.title + " - Showdown!" : "Showdown!";
 		},
 		updateAutojoin: function () {
-			if (Config.server.id !== 'showdown') return;
+			if (!Config.server.registered) return;
+			var curAutojoin = (Tools.prefs('autojoin') || '');
+			if (typeof curAutojoin === 'string' && Config.server.id !== 'showdown') {
+				// Switch to the autojoin object to handle multiple servers
+				curAutojoin = {showdown: curAutojoin};
+			}
```

Comment:
New lines 1651-1655 should be moved down to right above

    if (typeof curAutojoin !== 'string') {

Keeping everything together will make things more readable, and you'll probably be able to see other optimizations better, too.